Neos.Flow/Classes/Http/Client/CurlEngine.php

```diff
@@ -37,12 +37,17 @@ class CurlEngine implements RequestEngineInterface
 
     /**
      * Sets an option to be used by cURL.
+     * Setting headers is not allowed, please use request object.
      *
      * @param integer $optionName One of the CURLOPT_* constants
      * @param mixed $value The value to set
+     * @throws Http\Exception
      */
     public function setOption($optionName, $value)
     {
+        if ($optionName === CURLOPT_HTTPHEADER) {
+            throw new Http\Exception("Just CurlEngine specific options are allowed here. Please use request object for setting up request headers.", 1633334307);
```

Comment:
I suggest a more specific text such as
"Setting CURL headers is only possible via the request object, and not by using the setOption method"

Reply:
I'm also fine with that. But the generic `InvalidArgumentException` or a specifc one in the Flow/Http (`Neos\Flow\Http\InvalidArgumentException`) Package?

Reply:
I'd say the latter one.